source/common/upstream/upstream_impl.cc

```diff
@@ -533,6 +533,7 @@ bool BaseDynamicClusterImpl::updateDynamicHostList(const HostVector& new_hosts,
                                                    HostVector& hosts_added,
                                                    HostVector& hosts_removed, bool depend_on_hc) {
   uint64_t max_host_weight = 1;
+  bool eds_health_changed = false;
```

Comment:
If the host is already unhealthy, do we need to propogate that the host is now unhealthy for 2 reasons?
I'm assuming the answer is yes, but I'd like to understand why, and encourage you to comment in the code for future curious people.

Reply:
I can add this optimization, it's pretty straightforward.

Reply:
s/eds_health_changed/health_changed ?

Has the EDS health status changed of any endpoint? -> 
Has the EDS health status changed the health of any endpoint?